### PR TITLE
fix(query): Show privilege_escalation_allowed k8s alert also in case no securityContext is defined

### PR DESCRIPTION
**Pull request by Churro**
**Problem**

The `privilege_escalation_allowed` rule is rated high and, indeed, represents a very security-relevant setting. Currently, the rule does not trigger an alert in case no `securityContext` is defined with a container. There is a separate rule to check whether `securityContext` exists but the severity rating is low. Hence, alerts for a rule with high severity are only shown if a user acts upon a low severity rule. 

Therefore, I suggest to change the rule to always show the alert, regardless of the existence of `securityContext`. This is important since a report otherwise does not provide a full picture of issues that demand for action.

**Proposed Changes**

- Make the definition of `securityContext` optional, such that an alert is also shown in case `container.securityContext` is undefined

I submit this contribution under the Apache-2.0 license.

### CODE REVIEW

File: assets/queries/k8s/privilege_escalation_allowed/query.rego [hunk 1]
```diff
@@ -1,41 +1,43 @@
...
 	result := {
 		"documentId": input.document[i].id,
-		"searchKey": sprintf("metadata.name={{%s}}.%s.%s.name={{%s}}.securityContext.allowPrivilegeEscalation", [metadata.name, specInfo.path, types[x], containers[index].name]),
-		"issueType": "IncorrectValue",
-		"keyExpectedValue": sprintf("%s.%s[%s].securityContext.allowPrivilegeEscalation is false", [specInfo.path, types[x], containers[index].name]),
-		"keyActualValue": sprintf("%s.%s[%s].securityContext.allowPrivilegeEscalation is true", [specInfo.path, types[x], containers[index].name]),
+		"searchKey": sprintf("metadata.name={{%s}}.%s.%s.name={{%s}}.securityContext", [metadata.name, specInfo.path, types[x], container.name]),
+		"issueType": "MissingAttribute",
+		"keyExpectedValue": sprintf("metadata.name={{%s}}.%s.%s.name={{%s}}.securityContext.allowPrivilegeEscalation is set and is true", [metadata.name, specInfo.path, types[x], container.name]),
+		"keyActualValue": sprintf("metadata.name={{%s}}.%s.%s.name={{%s}}.securityContext.allowPrivilegeEscalation is undefined", [metadata.name, specInfo.path, types[x], container.name]),
```
Comment (rafaela-soares) [inline assets/queries/k8s/privilege_escalation_allowed/query.rego:41]:
```suggestion
		"keyExpectedValue": sprintf("metadata.name={{%s}}.%s.%s.name={{%s}}.securityContext.allowPrivilegeEscalation is set and is false", [metadata.name, specInfo.path, types[x], container.name]),
		"keyActualValue": sprintf("metadata.name={{%s}}.%s.%s.name={{%s}}.securityContext.allowPrivilegeEscalation is undefined", [metadata.name, specInfo.path, types[x], container.name]),
```

**Review by rafaela-soares — CHANGES_REQUESTED** [review]
Another great observation and refactor, @Churro 🚀 Thank you!

I have two request changes:

**Comment by Churro** [comment]
Suggestions applied, thank you for the thorough review @rafaela-soares! 👍

**Review by joaoReigota1 — CHANGES_REQUESTED** [review]
Hi @Churro, everything looks good to me, the only thing I would like to request here is the use of the searchLine

Thank you for your great contributions

**Comment by Churro** [comment]
Hi @joaoReigota1, thank you for the review. I've now added `searchLine` and another positive testcase related to that

**Review by joaoReigota1 — APPROVED** [review]
LGTM 🚀

**Review by rafaela-soares — APPROVED** [review]
LGTM 🚀

**Pull request merged**